target/linux/qualcommax/ipq50xx/base-files/lib/upgrade/glinet-upgrade.sh

```diff
@@ -0,0 +1,112 @@
+. /lib/upgrade/common.sh
+
+dumpimage_version="2025.01"
+
+glinet_validate_firmware() {
+	local type=$(identify_magic_long $(get_magic_long $1))
+	if [ "$type" = "unknown 73797375" ]; then type="sys"; fi
+
+	echo "$type"
+}
+
+glinet_do_fit_upgrade() {
+	echo -n "fit: Extract [ FIT IMAGE ] -x-x-> [ ubi.bin ] ... "
+	local ubi=/tmp/ubi.bin
+	local part=$(dumpimage -l "$1" | grep -o "Image [0-9] (ubi)" | cut -f2 -d" ")
+
+	local ubibin=$( dumpimage -T flat_dt -p ${part} -o "$ubi"  $1 )
+	if [ -s "$ubi" ]; then
+		echo "[ OK ]"
+		local ubiMd5=$(cat "$ubi" | md5sum | cut -f1 -d" ")
+		local ubi_size=$( cat "$ubi" | wc -c )
```

Comment:
Same here, you can simply use:
```suggestion
		local ubi_size=$(wc -c "$ubi")
```